### ISSUE TITLE
New error possibly related to new options in 3.0

### ISSUE THREAD

**Issue by benadamstyles** [jonathantneal/postcss-pseudo-class-enter#9]
Getting the following error:

```
Potentially unhandled rejection [5] TypeError: Cannot match against 'undefined' or 'null'.
  at module.exports.postcss.plugin (/Users/ben/Dev/Git/PROJECT/node_modules/postcss-pseudo-class-enter/index.js:6:63)
  at creator (/Users/ben/Dev/Git/PROJECT/node_modules/postcss/lib/postcss.js:150:35)
  at Function.get (/Users/ben/Dev/Git/PROJECT/node_modules/postcss/lib/postcss.js:159:27)
  at Processor.normalize (/Users/ben/Dev/Git/PROJECT/node_modules/postcss/lib/processor.js:136:12)
  at new Processor (/Users/ben/Dev/Git/PROJECT/node_modules/postcss/lib/processor.js:51:25)
  at PostCSS.postcss [as engine] (/Users/ben/Dev/Git/PROJECT/node_modules/postcss/lib/postcss.js:73:10)
  at PostCSS._render (/Users/ben/Dev/Git/PROJECT/node_modules/accord/lib/adapters/postcss/4.x - 5.x.js:35:24)
  at PostCSS.Adapter.render (/Users/ben/Dev/Git/PROJECT/node_modules/accord/lib/adapter_base.js:122:19)
```

I think this is to do with the new options you have added in `3.0` (which you don't mention in the changelog – caught me off guard!). I don't fully understand these options. Do you think you could elaborate a little in the readme? I don't understand what `prefix` is, or why the default is `'-'`, and I would like to know exactly what the `outline` option will output and why it is a string option instead of a boolean.

Thanks for all your help! I have been happily using this package for months so if I can help in any way, let me know.

**Comment by wesm87** [comment]
@Leeds-eBooks This error is happening because the updated plugin code assumes that an options object of some kind will be passed in. The code uses destructuring to extract the options out:

```
module.exports = postcss.plugin('postcss-pseudo-class-enter', ({
	outline = false,
	prefix = ''
}) => {
```

The problem is, if you don't pass an options object in, the function receives `undefined`, so what gets evaluated is something like the following:

```
const { outline = false, prefix = '' } = undefined
```

The quick fix is to just pass in an empty object for the options - `require('postcss-pseudo-class-enter')({})`.

**Comment by benadamstyles** [comment]
Ah ok, thanks. What about:
```
module.exports = postcss.plugin('postcss-pseudo-class-enter', ({
  outline = false,
  prefix = ''
} = {}) => {
```

**Comment by jonathantneal** [comment]
I am an hour or so from updating all my PostCSS plugins with safer destructuring. I finished updating all the plugins in need of this change last weekend, and I finished the postcss-partial-import update last night, and I'm making one more error handling improvement to postcss-tape. Once that's done, I can update all the dependencies, push the updates, wait 15 to be sure there were no errors from Travis (I run the tests locally first, too). When that all goes well, big ol' npm publish (on 20 odd plugins this would effect?)

**UPDATE**

- [X] postcss-tape is updated
- [X] postcss-partial-import is updated
- [x] Remaining PostCSS plugins are updated and pushed
- [x] Remaining PostCSS plugins pass tests and are published
